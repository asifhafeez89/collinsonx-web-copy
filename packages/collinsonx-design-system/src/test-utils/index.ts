import userEvent from '@testing-library/user-event';

export * from '@testing-library/react';
export { userEvent };
export { default as render, Provider } from './render';
