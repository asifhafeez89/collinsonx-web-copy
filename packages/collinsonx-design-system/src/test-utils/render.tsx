import { RenderOptions, render } from '@testing-library/react';
import { MantineProvider } from '@mantine/core';
import { ReactElement, ReactNode } from 'react';

export const Provider = ({ children }: { children: ReactNode }) => (
  <MantineProvider>{children}</MantineProvider>
);

type Render = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => ReturnType<typeof render>;

const customRender: Render = (ui, options) =>
  render(ui, {
    wrapper: Provider as unknown as RenderOptions['wrapper'],
    ...options,
  });

export default customRender;
