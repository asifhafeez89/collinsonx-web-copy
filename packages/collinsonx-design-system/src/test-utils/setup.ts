import userEvent from '@testing-library/user-event';
import render from './render';

export default function (jsx: Parameters<typeof render>[0]) {
  return {
    user: userEvent.setup(),
    ...render(jsx),
  };
}
