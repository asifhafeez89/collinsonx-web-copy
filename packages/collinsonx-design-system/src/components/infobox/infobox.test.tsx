import renderer from 'react-test-renderer';
import { Provider } from 'test-utils';
import Infobox from '.';

describe('<InfoBox />', () => {
  it('renders Infobox', () => {
    const tree = renderer
      .create(
        <Provider>
          <Infobox
            title="Your flight details"
            flight="Date"
            date="12/6/2023"
            handleEditClick={() => console.log()}
          />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
