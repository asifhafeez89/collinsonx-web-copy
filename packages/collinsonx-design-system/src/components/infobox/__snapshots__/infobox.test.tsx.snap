// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfoBox /> renders Infobox 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": "@media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}",
      }
    }
    data-mantine-styles="classes"
  />,
  .emotion-0 {
  border-radius: 10px;
  border: 1px solid #e9ecef;
  padding: 20px;
  margin-bottom: 10px;
  background: #ffffff;
}

<div
    className="emotion-0"
  >
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".__m__-r0{--grid-gutter:calc(0.125rem * var(--mantine-scale));}@media(min-width: 36em){.__m__-r0{--grid-gutter:var(--mantine-spacing-md);}}@media(min-width: 62em){.__m__-r0{--grid-gutter:var(--mantine-spacing-xl);}}@media(min-width: 88em){.__m__-r0{--grid-gutter:calc(1.25rem * var(--mantine-scale));}}",
        }
      }
      data-mantine-styles="inline"
    />
    <div
      className="m-410352e9 mantine-Grid-root __m__-r0"
      style={{}}
    >
      <div
        className="m-dee7bd2f mantine-Grid-inner"
        style={{}}
      >
        <style
          dangerouslySetInnerHTML={
            {
              "__html": ".__m__-r2{--col-flex-grow:1;--col-flex-basis:58.333333333333336%;--col-max-width:100%;}",
            }
          }
          data-mantine-styles="inline"
        />
        <div
          className="m-96bdd299 mantine-Grid-col __m__-r2"
          style={{}}
        >
          <style
            dangerouslySetInnerHTML={
              {
                "__html": ".__m__-r4{--grid-gutter:calc(0.125rem * var(--mantine-scale));}",
              }
            }
            data-mantine-styles="inline"
          />
          <div
            className="m-410352e9 mantine-Grid-root __m__-r4"
            style={{}}
          >
            <div
              className="m-dee7bd2f mantine-Grid-inner"
              style={{}}
            >
              <style
                dangerouslySetInnerHTML={
                  {
                    "__html": ".__m__-r6{--col-flex-grow:1;--col-flex-basis:100%;--col-max-width:100%;}",
                  }
                }
                data-mantine-styles="inline"
              />
              <div
                className="m-96bdd299 mantine-Grid-col __m__-r6"
                style={{}}
              >
                Your flight details
              </div>
              <style
                dangerouslySetInnerHTML={
                  {
                    "__html": ".__m__-r8{--col-flex-grow:1;--col-flex-basis:41.66666666666667%;--col-max-width:100%;}",
                  }
                }
                data-mantine-styles="inline"
              />
              <div
                className="m-96bdd299 mantine-Grid-col __m__-r8"
                style={{}}
              >
                <test-file-stub
                  h={24}
                  w={24}
                />
                 
                Date
              </div>
              <style
                dangerouslySetInnerHTML={
                  {
                    "__html": ".__m__-ra{--col-flex-grow:1;--col-flex-basis:58.333333333333336%;--col-max-width:100%;}",
                  }
                }
                data-mantine-styles="inline"
              />
              <div
                className="m-96bdd299 mantine-Grid-col __m__-ra"
                style={{}}
              >
                <test-file-stub
                  h={24}
                  w={24}
                />
                 
                12/6/2023
              </div>
            </div>
          </div>
        </div>
        <style
          dangerouslySetInnerHTML={
            {
              "__html": ".__m__-rc{--col-flex-grow:1;--col-flex-basis:8.333333333333334%;--col-max-width:100%;}",
            }
          }
          data-mantine-styles="inline"
        />
        <div
          className="m-96bdd299 mantine-Grid-col __m__-rc"
          style={{}}
        >
          <button
            className="mantine-focus-auto mantine-active m-8d3f4000 mantine-ActionIcon-root m-87cf2631 mantine-UnstyledButton-root"
            onClick={[Function]}
            style={
              {
                "--ai-bd": "calc(0.0625rem * var(--mantine-scale)) solid transparent",
                "--ai-bg": "var(--mantine-color-dark-6)",
                "--ai-color": "var(--mantine-color-white)",
                "--ai-hover": "var(--mantine-color-dark-7)",
                "position": "absolute",
                "right": 40,
                "top": 50,
              }
            }
            type="button"
          >
            <test-file-stub
              h={24}
              w={24}
            />
          </button>
        </div>
      </div>
    </div>
  </div>,
]
`;
